### PR TITLE
✨ Pin first message in thread

### LINKED ISSUE #56: 💡 Pin the first message in thread
Automatically pin the first message sent by bot in a thread so you can go to jump to it easily using pinned messages when you want to archive thread.

### PR DESCRIPTION
**Pull request by MarcusOtter**
Closes #56. No configuration - we assume people want this (otherwise just remove the manage messages permission for now).

### CODE REVIEW

File: src/handlers/messageHandler.ts [hunk 1]
```diff
@@ -124,10 +124,15 @@ async function autoCreateThread(message: Message, requestId: Snowflake) {
...
 	if (msgContent && msgContent.length > 0) {
-		await thread.send({
+		const msg = await thread.send({
 			content: msgContent,
 			components: [buttonRow],
 		});
+
+		if (botMember.permissionsIn(thread.id).has(Permissions.FLAGS.MANAGE_MESSAGES)) {
+			await msg.pin();
+			await thread.lastMessage?.delete();
```
Comment (c43721) [inline src/handlers/messageHandler.ts:134]: can you promse.all this please

**Review by c43721 — CHANGES_REQUESTED** [review]

File: src/handlers/messageHandler.ts (hunk 1 above)
Comment (MarcusOtter) [reply src/handlers/messageHandler.ts:134]: No, the pinned message needs to be created before we try to delete the latest message.

File: src/handlers/messageHandler.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@
 //
 // ________________________________________________________________________________________________
 
-import { type Message, MessageActionRow, MessageButton, NewsChannel, TextChannel, ThreadChannel, SnowflakeUtil, type Snowflake } from "discord.js";
+import { type Message, MessageActionRow, MessageButton, NewsChannel, TextChannel, ThreadChannel, SnowflakeUtil, type Snowflake, MessageAttachment, Permissions } from "discord.js";
```
Comment (nchristopher) [inline src/handlers/messageHandler.ts:18]: I assume this was VSCode being weird and importing something else.
```suggestion
import { type Message, MessageActionRow, MessageButton, NewsChannel, TextChannel, ThreadChannel, SnowflakeUtil, type Snowflake, Permissions } from "discord.js";
```

**Review by nchristopher — CHANGES_REQUESTED** [review]

File: src/handlers/messageHandler.ts (hunk 2 above)
Comment (MarcusOtter) [reply src/handlers/messageHandler.ts]: Good catch!

**Pull request merged**